specifications/api/swagger.yaml

```diff
@@ -1851,6 +1959,85 @@ components:
         state: *walletState
         tip: *blockReference
 
+    ApiActiveSharedWallet: &ApiActiveSharedWallet
+      type: object
+      required:
+        - id
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+        - balance
+        - delegation
+        - state
+        - tip
+      properties:
+        id: *walletId
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+        balance: *walletBalance
+        delegation: *ApiWalletDelegation
+        state: *walletState
+        tip: *blockReference
+
+    ApiPendingSharedWallet: &ApiPendingSharedWallet
+      type: object
+      required:
+        - id
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+      properties:
+        id: *walletId
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWallet: &ApiSharedWallet
+      type: object
+      oneOf:
+        - title: Pending shared wallet (collecting account public keys for each co-signer)
+          <<: *ApiPendingSharedWallet
+        - title: Active shared wallet (account public keys for each co-signers is collected)
+          <<: *ApiActiveSharedWallet
+
+    ApiSharedWalletPostData: &ApiSharedWalletPostData
+      type: object
+      required:
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+      properties:
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWalletPatchData: &ApiSharedWalletPatchData
+      type: object
+      required:
+        - cosigner
+        - account_public_key
+        - script_template_update
+      properties:
+        cosigner: *cosigner
+        account_public_key: *walletAccountPubkey
+        script_template_update: *scriptTemplateUpdate
```

Comment:
Instead of a body parameter here, I'd suggest to move that one to path parameters for a cleaner API. So instead of patching on `/shared-wallets/{walletId}`, we would patch on `/shared-wallets/{walletId}/payment-script-template` and `shared-wallets/{walletId}/delegation-script-template`. 

So this is consistent with the API for creating shared wallets and quite clear from the request itself. The request body then only includes the cosigner and the account public key. Of course, for adding a cosigners to both templates, 2 requests are needed; and it's okay :blush:

Reply:
done